server/raft_helpers_test.go

```diff
@@ -317,3 +317,34 @@ func (rg smGroup) waitOnTotal(t *testing.T, expected int64) {
 func newStateAdder(s *Server, cfg *RaftConfig, n RaftNode) stateMachine {
 	return &stateAdder{s: s, n: n, cfg: cfg, lch: make(chan bool, 1)}
 }
+
+func initSingleMemRaftNode(t *testing.T) (*raft, func()) {
+	t.Helper()
+	c := createJetStreamClusterExplicit(t, "R3S", 3)
+	s := c.servers[0] // RunBasicJetStreamServer not available
+
+	ms, err := newMemStore(&StreamConfig{Name: "TEST", Storage: MemoryStorage})
+	require_NoError(t, err)
+	cfg := &RaftConfig{Name: "TEST", Store: t.TempDir(), Log: ms}
+
+	err = s.bootstrapRaftNode(cfg, nil, false)
+	require_NoError(t, err)
+	n, err := s.initRaftNode(globalAccountName, cfg, pprofLabels{})
+	require_NoError(t, err)
+
+	cleanup := func() {
+		c.shutdown()
+	}
+	return n, cleanup
+}
+
+// Encode an AppendEntry.
+// An AppendEntry is encoded into a buffer and that's stored into the WAL.
```

Comment:
I don't get the reference to the WAL here.
Looks to me like this is pure side-effect helper: attach a buffer to the AE which contains a serialized version of the AE itself.

Reply:
The AppendEntry needs to be encoded into a buffer, and that buffer is what is stored into the WAL (outside of this encoding part). Without encoding we will not be able to store it.

Could maybe rephrase it to make that clearer?